savoirfairelinux/sous-chef
# Fixed #689: Add override confirmation to "New Orders"

## Fixes #689 

### Changes proposed in this pull request:

* Adds an override confirmation modal to the Orders > New Orders page when an order already exists for a selected day.

### Status


### How to verify this change

* Go to Orders, click New orders
* Select a user name
* Select a day with an existing order
* A modal should appear informing you that an order exists for this user on the selected date
  * Selecting "Cancel" should remove the date in question from the selected dates
  * Selecting "Edit Existing" should redirect to the Update Order page for the existing order
  * Selecting "Override" should allow the user to continue, and add the date to a hidden "override dates" array. Upon completing the order process, the existing order should be deleted, and the new order created in its place.

### Deployment notes and migration


### New translatable strings


### Additional notes

* The existing order object in the DB is deleted when the user chooses to override it.
* The user is only notified of an existing order if that order is not cancelled.
* If the order has a different status other than simply "Ordered", this status will be shown to the user in the confirmation modal.
